src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1510,6 +1529,39 @@ abstract class RefChecks extends Transform {
         currentApplication = tree
         tree
     }
+
+    private object StringContextIntrinsic {
+      def unapply(t: Apply): Option[(List[Tree], List[Tree])] = {
```

Comment:
Could return an `Option[(List[Literal], List[Tree])]` to avoid the casts at the use site, though it would require another `map` in the `raw` case. Up to you, both ways is fine.